module2-database-technologies/README.md

```diff
@@ -1 +1,372 @@
-# curriculum-backend-readings
\ No newline at end of file
+# Module 2: Databases
+
+## Introduction to Databases
+In this module, you'll learn about **databases**. Database is a general term
+that refers to the data stored in a structured format. We use databases when we
+want to persist data for future use. For example, in a web application, when a
+user writes something in a text box, this data will be lost when the page is
+refreshed -- unless this data is persisted in a database.
+
+Databases are a ubiquitous and foundational part of backend technology.
+
+### Introduction to relational databases
+
+One of the most common types of databases is the **relational database**. A
+relational database stores data in tables, and these tables may have *relations*
+to one another. For example, you may have a table `Customer`; each row
+represents a customer in your application. You may have another table 
+`Order`, a customer can order many things, and each order will appear in this
+table. We will later look in more detail at how the two tables are linked.
+
+Relational databases are frequently referred to as SQL databases. SQL is a
+[declarative programming
+language](https://365datascience.com/tutorials/sql-tutorials/sql-declarative-language/)
+that is commonly used to query for data. Not all relational databases
+necessarily use SQL, but it is by far the most common, so in practice, SQL
+database and relational database are synomyous terms.
+
+The most frequently used relational databases are PostgreSQL and MySQL. There
+are many offerings for SQL databases, and each may have slightly different
+syntax for SQL. However, if you understand the general concept of relational
+databases, you will be able to easily use different databases. 
```

Comment:
Can we share some more details on MySQL and PostgreSQL? Such as how long has it been around, what are the general advantages & disadvantages, which tech stack is it mostly used in. For example, MySQL is most commonly used along with PHP or in the LAMP stack. Similarly, how PostgreSQL supports both relational and non-relational querying.
I think [this codecademy article](https://www.codecademy.com/articles/what-is-rdbms-sql) that you had added on the research library covers these pieces.

Reply:
what do you mean exactly by non-relational querying? like a JSON type? because from my googling that's available in most of the big offerings

in general i think it would be more misleading for students to think that there's any reason you'd pick one over the other except perhaps for convenience if you were installing some stack directly that came with MySQL, because for any use case a student would use i don't think there would be any reason to choose one over the other

Reply:
Yup JSON should be supported by all offerings. I think this is what I meant: https://www.enterprisedb.com/blog/8-ways-rdbms-postgres-handles-non-relational-data
However, zooming out the idea wasn't to focus on certain advantages or disadvantages of each RDBMS so that students would choose one over the other. But more to explain why so many different offerings exist in the first place. Students would focus on MongoDB in module 3 for sure, but we want them to be comfortable with agnostic approach so that tomorrow if they don't work with MongoDB they know why or at least know how to look for the answers behind the company's choice.